tests/unit/loaders/test_pdf_loader.py

```diff
@@ -22,7 +22,7 @@ def test_load(self, loader):
 
         artifacts = loader.load(path)
 
-        assert len(artifacts) == 149
+        assert len(artifacts) == 151
```

Comment:
Have not dug into why this changed. I assume it's from some edge-case in the PDF chunking that is now handled by the non-separator fallback method.

Reply:
Just thought of something...let's add tests for the last chunk that tests the last sentence with `endswith`, so are 100% sure that we are not dropping any chunks in the process.

Reply:
Done